tests/nodeos_forked_chain_test.py

```diff
@@ -375,6 +375,8 @@ def getBlock(self, blockNum):
     if preKillBlockProducer == "defproducerj" or preKillBlockProducer == "defproducerk":
         # wait for defproduceri so there is plenty of time to send kill before defproducerk
         nonProdNode.waitForProducer("defproduceri")
+        preKillBlockNum=nonProdNode.getBlockNum()
+        preKillBlockProducer=nonProdNode.getBlockProducerByNum(preKillBlockNum)
```

Comment:
Can we move those two assignments (`preKillBlockNum` and `preKillBlockNum`) out of the `if`, and delete the same assignments on line 373 and 374?

Reply:
Honestly, could probably get rid of 373-375 and just make the `waitForProducer` unconditional and get the same result.

Reply:
I don't see how. `preKillBlockNum` is used in `nonProdNode.getBlockProducerByNum(preKillBlockNum)` call so it is needed in both places as the value has changed.

Reply:
If you always wait that adds a very long wait to the test in a lot of cases it is not needed.

Reply:
You're right. I was thinking about that a little incorrectly -- forgetting how many producers were enabled here.  Yeah, I think it is probably fine the way it is, as you need the assignments on 373-4 to use in the conditional 375, and only reassign if proper conditions met.

Reply:
Still do the wait inside the `if`. But calculate preKillBlockNum and preKillBlockProducer after the `if`, so the calculation is only done once.

Reply:
The `if` uses `preKillBlockProducer` which is obtained from `nonProdNode.getBlockProducerByNum(preKillBlockNum)` which needs `preKillBlockNum`.

Reply:
`preKillBlockProducer` is used in the `if`, how are you thinking about getting around that?

Reply:
Oha, I missed preKillBlockNum is used in the  `if` condition. Sorry.